thoth/package_extract/core.py

```diff
@@ -47,20 +50,31 @@ def extract_buildlog(input_text: str) -> typing.List[dict]:
 def extract_image(image_name: str, timeout: int = None, *, registry_credentials: str = None,
                   tls_verify: bool=True) -> dict:
     """Extract dependencies from an image."""
-    image_name = quote(image_name)
-    with tempfile.TemporaryDirectory() as dir_path:
-        download_image(
-            image_name,
-            dir_path,
-            timeout=timeout or None,
-            registry_credentials=registry_credentials or None,
-            tls_verify=tls_verify
-        )
+    """Begin the timer for when the job starts"""
+    with _METRIC_ANALYZER_JOB.time():
+            image_name = quote(image_name)
+            with tempfile.TemporaryDirectory() as dir_path:
+                download_image(
+                    image_name,
+                    dir_path,
+                    timeout=timeout or None,
+                    registry_credentials=registry_credentials or None,
+                    tls_verify=tls_verify
+                )
 
-        rootfs_path = os.path.join(dir_path, 'rootfs')
-        layers = construct_rootfs(dir_path, rootfs_path)
+                rootfs_path = os.path.join(dir_path, 'rootfs')
+                layers = construct_rootfs(dir_path, rootfs_path)
 
-        result = run_analyzers(rootfs_path)
-        result['layers'] = layers
+                result = run_analyzers(rootfs_path)
+                result['layers'] = layers
 
-        return result
+                return result
```

Comment:
Isn't the result returned too soon? It should be after the push I guess.

Reply:
I tried testing it with the result being returned before and after the push, the run time of the job remained the same. But to be safe I will return it after the push.